src/nni_manager/core/nnimanager.ts

```diff
@@ -135,6 +135,42 @@ class NNIManager implements Manager {
         return this.dataStore.storeTrialJobEvent('ADD_CUSTOMIZED', '', hyperParams);
     }
 
+    public async resubmitTrialJob(data: string): Promise<void> {
+        // get trail job info by jobId
+        const trialJobId = JSON.parse(data).job_id
+        const trialJobInfo = await this.getTrialJob(trialJobId); 
+        this.log.debug(`trialJobId: ${trialJobId}`);
+        
+        if (this.dispatcher === undefined) {
+            return Promise.reject(
+                new Error('tuner has not been setup')
+            );
+        } 
+        if (trialJobInfo.hyperParameters === undefined) {
+            return Promise.reject(
+                new Error('no hyper parameters exist')
+            );
+        } 
+
+        let hyperParameter
+        if (trialJobInfo.hyperParameters.length == 1){
+            hyperParameter = trialJobInfo.hyperParameters[0]
+        } else {
+            hyperParameter = trialJobInfo.hyperParameters.find(hp=>JSON.parse(hp).parameter_index == 0) as string
+        }
+
+        if(trialJobInfo.status != 'FAILED'){
+            return Promise.reject(
+                new Error('only failed trials can be resubmitted')
+            );
+        }
```

Comment:
This status checking can be moved up before retrieving hyperParameter.

Reply:
fixed